internal/app/go-filecoin/connectors/storage_market/client.go

```diff
@@ -229,5 +239,50 @@ func (s *StorageClientNodeConnector) GetChainHead(ctx context.Context) (shared.T
 }
 
 func (s *StorageClientNodeConnector) OnDealSectorCommitted(ctx context.Context, provider address.Address, dealID abi.DealID, cb storagemarket.DealSectorCommittedCallback) error {
-	panic("implement me") // @laser to do this
+	view, err := s.chainStore.StateView(s.chainStore.Head())
+	if err != nil {
+		cb(err)
+		return err
+	}
+
+	resolvedProvider, err := view.InitResolveAddress(ctx, provider)
+	if err != nil {
+		cb(err)
+		return err
+	}
+
+	err = s.waiter.WaitPredicate(ctx, func(msg *types.SignedMessage, c cid.Cid) bool {
+		if msg.Message.To != provider {
```

Comment:
This doesn't do as you'd hope. The on-chain message could have either the resolved or the unresolved address. You need to check both. Or don't at all.

Reply:
It now resolves both addresses